### ISSUE TITLE
Consider adopting a Code of Conduct

### ISSUE THREAD

**Issue by mohe2015** [quicklisp/quicklisp-projects#1942]
https://opensource.guide/code-of-conduct/ has detailed information.
I just would like to discuss this and also know the maintainers' opinions.
I posted it in this repo as there doesn't seem to be any main repository.

Edit: To be clear I don't prefer any outcome in the current case. I would just like it if there would be clear rules which explain the decisions. But it also would mean more responsibility for the maintainers as they would then need to also enforce it properly.

Also a Code of Conduct may rule in favor of Hexstream considering the recent comment in the other issue.

**Comment by thephoeron** [comment]
An official code of conduct is a helpful thing to add to every open-source project. It's a good place for maintainers to state their expectations.

**Comment by Harleqin** [comment]
The question is: a code of conduct for whom?

- Authors of libraries?  I don't think they even need to know about the existence of Quicklisp.  Anyone can request inclusion of any library.  Why should they be bound or care?

- Authors of requests?  Yeah, but they will be handled anyway through the usual moderation tools, no?  Does it make sense to ignore an issue just because the author of the bug ticket misbehaves?  What are the consequences of misbehaving except moderation of posts (which happens anyway)?

- Downloaders?  They are completely anonymous, and there is little room to even misbehave in the course of downloading.

In conclusion, I believe that Quicklisp does not _have_ social interaction, especially since inclusion or exclusion of a library is done on purely technical considerations.

**Comment by mohe2015** [comment]
@Harleqin I would assume that any communication in the quicklisp organizations issues and pull requests (and contents) would need to follow the Code of Conduct. If there is an official IRC or so it also should. Of course the scope is pretty limited but I still think it would be good practice.

**Comment by thephoeron** [comment]
@Harleqin: I'm not certain that those criticisms are valid or relevant. A code of conduct governs the way contributors treat each other, while a license governs access and usage of the software itself. Moderators are rarely held accountable to their community, and that's a problem. Furthermore, belief is an intrinsically fallacious preconception of truth resulting from a confusion of terms, which has no bearing on reality, and the very act of responding to this thread to play contrarian is a contradiction of your own statement of belief, which is a purely social interaction of no technical value designed to gratify one's own Ego at the community's expense. So... thank you for the satisfiable demonstration that Quicklisp requires a code of conduct.

**Comment by Harleqin** [comment]
@thephoeron I feel that you really went too far with the part of your post starting with “furthermore”.  I can offer you to delete this post if you delete that.  I really am hurt.

When I say “I believe”, it is to indicate that I am expressing my current view, and that I am ready to change it.  You constructing an ego trip out of it is really not warranted, and it feels to me like you are trying to attack not the argument, but the fact that an argument was made at all.  Again, I may misrepresent what you really try to do, and maybe my use of “believe” is not what you are used to.  I guess that we'd need to synchronize our understanding of the english language (which is not my mother tongue, by the way, so it may well be that it is I who has to learn the connotations people will generally assume when reading “believe”).

I'll return to the actual issue in the next post.

**Comment by thephoeron** [comment]
@Harleqin: I didn't intend any spite whatsoever. I really like logic and debate, and if I offer a rebuttal, it's only targeted at the argument and not the person making it. Sometimes, I throw in a fallacy myself to spice up a debate. But proof by contradiction of the antithetical is one of the strongest proof tactics.

**Comment by Harleqin** [comment]
OK, I can see that a code of conduct can be meaningful with regard to the posts (in issues) and other interactions with a repository or set of repositories.

To elaborate, I do not think that every repository needs a code of conduct, because I'd assume that anyone posting issues, answers, pull requests etc. is aware that they are acting in the sphere of authority of that repositories owner.

In the case of a repository or set of repositories that mainly serve a community (this is a bit vague but I hope you see the distinction I am trying to make here) and therefore has a high rate of interaction with the members of that community, a code of conduct can set the expectations so that there is a feeling of “rule of law”, instead of being at the whim of the repo owner.

What I do not see is that a code of conduct can pertain to activities not directly related to these repos (i. e. the Quicklisp repos).

**Comment by Harleqin** [comment]
@thephoeron I'd appreciate you keeping your spice to yourself.  It is hard enough to keep a debate to the facts without it.

**Comment by thephoeron** [comment]
@Harleqin: oddly enough, that's exactly the situation it's for (when a debate or thread rabbit-holes). It's meant as a way to signal suppressed rational faculties when emotions take over a conversation, like when the Ego gets in the way, or when debates turn into arguments that get truly horrible.

The idea of a code of conduct is, we should be free to say whatever needs to be said, and be empowered to speak our mind. Attacking the person for opening up is toxic, while debating an idea constructive and supportive. Censorship and secrecy is bad business, and it's better when everyone is held accountable. Too much safety and comfort and people begin to identify with their ideology instead of remembering it is a product of their imagination, and that the power of their mind is effectively infinite.

**Comment by Harleqin** [comment]
@thephoeron I perceived your interjections so far as attempts to derail the discussion or demerit arguments by blowing up single words into meta-meta arguments about general fallacies of the mind.  If this was a personal repo of mine, I'd probably use the moderation tools to filter that out — although I might open a parallel meta-issue if I see value in what is being said, just not where it is said.  In a community repo (whatever that is, see above), I'd probably try to express the decision process for that in a code of conduct (among other things).

I do understand that you do not _intend_ derailing or baseless demeriting, but seem to genuinely believe that your apparently philosophical detours do further the discussion.  However, I think that you are putting too much weight on single words and have a bias of perceiving ideology in others.

You may have noticed that I already shifted my position after @mohe2015's post.  Unless you were too distracted by your perception of my general shortcomings in the words I was writing.

**Comment by K1D77A** [comment]
Don't fix what aint broke, simple az.

**Comment by thephoeron** [comment]
@Harleqin I take it you're a really unhappy person, and habitually find insult where there is none. I'm pretty happy-go-lucky. We're bound to misunderstand each other.

However. Everything I've brought up is either directly relevant to a code of conduct, or in direct support of a novel perspective on the code of conduct, which I can safely presume needs elaboration for most readers without waiting for someone to ask.

You also completely ignored my framing of codes of conduct in general, in order to go on about my behavior, while actually describing your own. So I'd like to call attention to passive-aggressive shaming tactics and projection, which are very much intentional, and must definitely be highlighted for inclusion in any code of conduct to be considered.

**Comment by Harleqin** [comment]
@thephoeron I have no interest in a “no you're bad — no you're bad” debate, which I could now continue.  My offer stands to delete each of my following posts:

- https://github.com/quicklisp/quicklisp-projects/issues/1942#issuecomment-735319361
- https://github.com/quicklisp/quicklisp-projects/issues/1942#issuecomment-735322294
- https://github.com/quicklisp/quicklisp-projects/issues/1942#issuecomment-735379778
- and this one,

if you agree to remove the following:

- The part of https://github.com/quicklisp/quicklisp-projects/issues/1942#issuecomment-735317058 starting with “Furthermore”
- https://github.com/quicklisp/quicklisp-projects/issues/1942#issuecomment-735320323
- https://github.com/quicklisp/quicklisp-projects/issues/1942#issuecomment-735338577
- https://github.com/quicklisp/quicklisp-projects/issues/1942#issuecomment-741747835

because I think that the conversation in them is off topic for this thread.

**Comment by thephoeron** [comment]
@Harleqin I don't agree that it is such a debate. I'm not judging you—I only have a few observations of behaviour in textual form, where most of the context and nuance of natural languages are never encoded. All I can do is critique behaviour, which is not _you_, it is just a series of actions you _chose_, or which you were trained to do automatically without needing to think, such as walking.

We are discussing a code of _conduct_, correct behaviour so we all feel welcome, heard, empowered to stand up and share informed opinion to spark debate—but not to proselytize or dictate or insult each other. My aggressive, integrated, and improvisational sense of humour was not appreciated by you? No problem. Happy to apologize. Strangely, you see an apology as weakness and mounted a personal attack, which I find a bit odd. Normally only psychopaths do that. But I've never met anyone as unhappy as you, so I simply don't have the context to deconstruct your intentions.

I also don't agree that any of these messages are really, truly, completely off-topic. We are discussing codes of conduct, and having a hard time getting along in the process.

Our messages _really_ matter. The result of this conversation will either serve as a standard to follow, or a warning to heed.

Would you like peace, or would you rather try and win?

**Comment by nibbula** [comment]
This whole discussion about codes of conduct should be permabanned given the principles in the code of conduct, including this comment.

**Comment by comradekingu** [comment]
@thephoeron 
> A code of conduct governs the way contributors treat each other,

I should hope not.

**Comment by thephoeron** [comment]
@comradekingu Oh? What is it that you hope for?

The natural inverse implied by weakest commitment, is that contributors need not be governed, that any behaviour is acceptable.

By strongest commitment, that contributors should not be governed, especially not by mutual agreement, and all behaviour is self-justified.

Or did you want to correct the bounds on the domain of discourse before we proceed?

**Comment by comradekingu** [comment]
I disagree two-fold there.
There is no natural state of having a CoC, so its omission is neither here nor there.
It governs behavior, attemptedly. If a strong governor of how contributors treat each-each other, it would pose quite a risk to get it wrong. Not that I think people even believe this. This is where the inverse implication truly lies.
Do projects without them do worse, or better? Do projects do better or worse after getting one?

Imagine someone antisocial enough to have their behavior written down as a warning to others, would modify any of it because it sits in a textfile nobody reads. If they do, there is false security to be had for this reason, as those aren't the same people. If we stay within the realm of what few other people than psychopaths do, why isn't a CoC system their playground? Every conflict escalated beyond resolving problems between people, dealt with in secret, without any trifecta of power or facing ones tribunal or accuser. At any point there is as great a potential for malice, only that part is by the letter and often comes with named positions. It can be small changes from one version to the next. Which ones have you read?

Moreover, most of them allow invariant sections, so it means reading it again and again, for every project that has one. Not that people don't actively avoid contributing if there is one, but what upside is there to having one?

The various different ones are all different enough to be mutually exclusive, so why the urge to have one when it means siding with _one_ particular brand of forever changing revisions, or so outdated it is indistinguishable from satire?

What is it they have done that couldn't already be done, and would have to be done anyway? The promise is great in its vagueness, and the spelled out and detailed authoritarianism is just accepted as a result. I am of course a staunch supporter of meritocracy, and thus am irrelevant. Yet I wonder, what is the point?

> Moderators are rarely held accountable to their community, and that's a problem.

Do you envision enforcing any CoC without those? If it is really just text that people don't read doing it, why are laws backed by a monopoly on violence not really doing it?
If the problem is maintainers have a vested interest and thus hold this power, how does it sound better without vested interest or aspiration to attain it? You are at any rate still dicing up power into some hands, and calling it a community.

I really don't understand. How would having a CoC help this discussion for example? What would it say? What would be done about it?

**Comment by thephoeron** [comment]
@comradekingu Great answer. You've voiced significant concerns that we all would be wise to remember. And here I thought you were only trolling. Sorry about that.

This thread started because we had another wild blow-up that was going full holy war. It just kept escalating.

I'm not interested in telling people what to do or how to do it. I was unhappy about having to insinuate myself in the mentioned blow-up. I'm also not a huge fan of handling people like fragile ornaments. I want a Lisp community where everyone feels welcome, can be themselves, and respect each other enough to challenge each other and be challenged. People disagree, and that's normally a good thing when they talk it through.

The issue I see, and the underlying issue of your concerns, I think are one and the same. Imposition is always the problem. Right?

**Comment by K1D77A** [comment]
I am not a fan of CoC's I never have been because I often end up on the receiving end of the "moderators" who are rarely if ever "moderate". A CoC is simply opening the door for a few Machiavellian types to determine the behaviour of those who are just going along to get along and I don't want the behaviour of those who don't care to get involved in the power process dictated by the types who would be attracted to such roles. Long Live Xach, King of Quicklisp!

**Comment by Hexstream** [comment]
Yeah, let's stick with dangerous scam artists doxing top Common Lisp Open Source contributors and scamming twitter into locking their account for calling out their doxing without any negative repercussions to them whatsoever, instead. Why rock the boat when everything's already perfectly fair for everyone?

**Comment by K1D77A** [comment]
I agree. Don't rock the boat.

**Comment by thephoeron** [comment]
If you find yourself constantly on the receiving end of moderation, it isn't a Code of Conduct or the moderators that are the problem. It is the consistent behaviour pattern that is the problem, because it is intrinsically toxic.

I've been in this position myself, courtesy of PTSD, but in meatspace. Found myself caught between two psychopaths, a Machiavellian and a Classic Narcissist, who occasionally collaborated when it allowed them to serve their own interests. No one could deal with the level of psychopathic abuse I was facing, so I stopped talking about it, took a deep breath, and fixed the problem on my own by refusing to react and turning them against each other. Changing my own behaviour didn't stop anything they were doing, it made them both escalate. But then I no longer had to ask for help, because they gave themselves away.

The best thing about Ethics is that it is self-correcting.

So I would just like to point out one thing. Hexstream is the instigator that spurred the need for this conversation. He's a psychopath, too. Escalating. Again. So you might want to learn to swim.

**Comment by Hexstream** [comment]
Almost correct, except I am not the instigator and not a psychopath, and constantly pointing out blatant injustices is not "escalating".

Whenever you think of blaming the victim: don't.

If you really want to get at the truth, then listen to people who constantly tell the truth, not the slanderers.

Anyway, I don't see that the Common Lisp community might be anywhere near salvageable for the time being, so we might as well spare ourselves yet another fruitless struggle session.

Ultimately, the Common Lisp community will live or die by infrastructure. I hope I don't need to tell you that right now it is almost completely dead, and still going downhill.

**Comment by comradekingu** [comment]
@Hexstream What is this textfile going to do to that effect?
GitHub has a CoC, and I am sure Twitter has one too.

**Comment by Hexstream** [comment]
Regulate user and moderator behavior.

[Contributor Covenant](https://www.contributor-covenant.org/) seems pretty good as a baseline.

**Comment by K1D77A** [comment]
"Found myself caught between two psychopaths, a Machiavellian and a Classic Narcissist"
Thanks for demonstrating my point.

> Regulate user and moderator behavior.
> 
> [Contributor Covenant](https://www.contributor-covenant.org/) seems pretty good as a baseline. There's a [FAQ](https://www.contributor-covenant.org/faq/).

Seems most of us are capable of regulating our own behaviour and resolving disputes in a reasonable way, this is totally unnecessary.

**Comment by Hexstream** [comment]
> Seems most of us are capable of regulating our own behaviour and resolving disputes in a reasonable way

Yeah, and then a dangerous scam artist comes along and posts a series of scams misrepresenting a top Common Lisp Open Source contributor as "not a Common Lisp expert", makes a twitter just to harass him, doxes him, scams twitter into locking his account for calling out his doxing, and then suddenly [the Common Lispiest twitter since July 2012](https://twitter.com/HexstreamSoft) is apparently locked forever, for no reason whatsoever, and the filthy scam artist and doxer gets away with it with no repercussions to him whatsoever forever.

I believe that person has thoroughly demonstrated that they are incapable of "regulating [their] own behaviour and resolving disputes in a reasonable way", hence we could use some guardrails instead of going off the rails.

**Comment by thephoeron** [comment]
Oh well. This was a lovely and stimulating conversation while it lasted.

I'm out.

**Comment by ghollisjr** [comment]
This thread is sufficient proof to me that quicklisp should remain under the benevolent dictatorship of Zach without interference from committee.

Codes of conduct are like rope.  You can use rope to tie down a heavy load on the back of a truck so it gets to its destination safely.  You can also use rope to tie someone to railroad tracks.  In my experience, when you already have sound direction from someone with vested interest in the success of an enterprise, adding legalese disguised as community safety measures both attracts the wrong kinds of attention as well as hands bad actors free rope to tie up whomever they wish.  It starts with stopping a few trolls and it ends with smothering a project.

**Comment by Hexstream** [comment]
@mohe2015 Thank you for the suggestion!

@Harleqin Thank you for your pertinent comments.

**Comment by comradekingu** [comment]
And then a person comes along and posts an appeal to rules a scam artist will have _serious_ qualms breaking, in light of doing the things supposedly in need of being pointing out. Because. There is also zero way a dangerous scam artist wouldn't be thwarted by such guidelines, and no way said dangerous scam artist would use them to own advantage.

"regulating [their] own behaviour and resolving disputes in a reasonable way" is great language for a CoC, hence putting guardrails in place(?), rather than being off the rails already. Who knows.
